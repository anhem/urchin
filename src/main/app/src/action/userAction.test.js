import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
import {
    addGroup,
    createUser,
    deleteUser,
    getGroupsForUser,
    getUser,
    getUsers,
    removeGroup,
    setUser
} from "./userAction";

describe('userAction', () => {

    const getUsersResponse = {};
    const folderId = 1;
    const groupId = 10;
    const userId = 100;
    const user = {username: 'user'};
    let store;

    beforeEach(() => {
        store = configureMockStore([thunk])({});
    });

    afterEach(() => {
        fetchMock.reset();
        fetchMock.restore();
    });

    it('getUsers is successful', () => {

        fetchMock.getOnce('/api/users', getUsersResponse);

        return store.dispatch(getUsers()).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('getUser is successful', () => {

        fetchMock.getOnce('/api/users/' + userId, getUsersResponse);

        return store.dispatch(getUser(userId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('setUser is successful', () => {

        store.dispatch(setUser(user));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('createUser is successful', () => {

        fetchMock.postOnce('/api/users/add', user);

        return store.dispatch(createUser(user)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('createUser failed', () => {

        return store.dispatch(createUser(user)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('deleteUser is successful', () => {

        fetchMock.deleteOnce('/api/users/' + userId, {});

        return store.dispatch(deleteUser(userId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('deleteUser failed', () => {

        return store.dispatch(deleteUser(userId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('getGroupsForUser is successful', () => {

        const getGroupsResponse = {};

        fetchMock.getOnce('/api/users/' + userId + "/groups", getGroupsResponse);

        store.dispatch(getGroupsForUser(userId));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('addGroup is successful', () => {

        const getGroupsResponse = {};
        const data = {
            userId: userId,
            groupId
        };

        fetchMock.postOnce('/api/groups/user', data);
        fetchMock.getOnce('/api/users/' + userId + "/groups", getGroupsResponse);

        return store.dispatch(addGroup(userId, groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('addGroup failed', () => {

        return store.dispatch(addGroup(userId, groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('removeGroup is successful', () => {

        const getGroupsResponse = {};
        fetchMock.deleteOnce('/api/groups/' + groupId + "/user/" + userId, {});
        fetchMock.getOnce('/api/users/' + userId + "/groups", getGroupsResponse)

        return store.dispatch(removeGroup(userId, groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('removeGroup failed', () => {

        return store.dispatch(removeGroup(userId, groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

});