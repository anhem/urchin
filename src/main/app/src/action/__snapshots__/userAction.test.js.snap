// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userAction addGroup failed 1`] = `
Array [
  Object {
    "type": "ADD_GROUP_TO_USER",
  },
]
`;

exports[`userAction addGroup is successful 1`] = `
Array [
  Object {
    "type": "ADD_GROUP_TO_USER",
  },
  Object {
    "data": Object {
      "groupId": 10,
      "userId": 100,
    },
    "type": "ADD_GROUP_TO_USER_SUCCESS",
  },
  Object {
    "type": "GET_GROUPS_FOR_USER",
  },
  Object {
    "data": Object {},
    "type": "GET_GROUPS_FOR_USER_SUCCESS",
  },
]
`;

exports[`userAction createUser failed 1`] = `
Array [
  Object {
    "type": "SAVE_USER",
  },
]
`;

exports[`userAction createUser is successful 1`] = `
Array [
  Object {
    "type": "SAVE_USER",
  },
  Object {
    "data": Object {
      "username": "user",
    },
    "type": "SAVE_USER_SUCCESS",
  },
]
`;

exports[`userAction deleteUser failed 1`] = `
Array [
  Object {
    "type": "DELETE_USER",
  },
]
`;

exports[`userAction deleteUser is successful 1`] = `
Array [
  Object {
    "type": "DELETE_USER",
  },
  Object {
    "data": Object {},
    "type": "DELETE_USER_SUCCESS",
  },
]
`;

exports[`userAction getGroupsForUser is successful 1`] = `
Array [
  Object {
    "type": "GET_GROUPS_FOR_USER",
  },
]
`;

exports[`userAction getUser is successful 1`] = `
Array [
  Object {
    "type": "GET_USER",
  },
  Object {
    "data": Object {},
    "type": "GET_USER_SUCCESS",
  },
]
`;

exports[`userAction getUsers is successful 1`] = `
Array [
  Object {
    "type": "GET_USERS",
  },
  Object {
    "data": Object {},
    "type": "GET_USERS_SUCCESS",
  },
]
`;

exports[`userAction removeGroup failed 1`] = `
Array [
  Object {
    "type": "REMOVE_GROUP",
  },
]
`;

exports[`userAction removeGroup is successful 1`] = `
Array [
  Object {
    "type": "REMOVE_GROUP",
  },
  Object {
    "data": Object {},
    "type": "REMOVE_GROUP_SUCCESS",
  },
  Object {
    "type": "GET_GROUPS_FOR_USER",
  },
  Object {
    "data": Object {},
    "type": "GET_GROUPS_FOR_USER_SUCCESS",
  },
]
`;

exports[`userAction setUser is successful 1`] = `
Array [
  Object {
    "data": Object {
      "username": "user",
    },
    "type": "SET_USER",
  },
]
`;
